File: services/agent-engine/app/schemas/org.py
"""Schemas for the org chart (who manages whom)."""

from __future__ import annotations

from pydantic import BaseModel


class OrgAgent(BaseModel):
    id: str
    name: str
    emoji: str = ""
    model: str = ""
    workspace: str = ""
    is_default: bool = False


class OrgLinkItem(BaseModel):
    agent_id: str
    parent_agent_id: str = ""  # empty → top level (no manager)


class OrgChartResponse(BaseModel):
    ok: bool
    message: str = ""
    agents: list[OrgAgent]
    links: list[OrgLinkItem]


class OrgSaveRequest(BaseModel):
    links: list[OrgLinkItem]


class OrgActionResult(BaseModel):
    ok: bool
    message: str = ""

File: apps/desktop/src/renderer/src/lib/api/types/org.ts
/**
 * Org chart — who manages whom. Mirrors `services/agent-engine/app/schemas/org.py`.
 */

export interface OrgAgent {
  id: string;
  name: string;
  emoji: string;
  model: string;
  workspace: string;
  is_default: boolean;
}

export interface OrgLinkItem {
  agent_id: string;
  /** Empty string → top level (no manager). */
  parent_agent_id: string;
}

/** Response from `GET /org/chart`. */
export interface OrgChartResponse {
  ok: boolean;
  message: string;
  agents: OrgAgent[];
  links: OrgLinkItem[];
}

/** Response from `PUT /org/chart`. */
export interface OrgActionResult {
  ok: boolean;
  message: string;
}
